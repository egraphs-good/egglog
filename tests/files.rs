use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
