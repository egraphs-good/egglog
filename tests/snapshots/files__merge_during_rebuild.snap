---
source: tests/files.rs
expression: serialized
---
nodes:
  x-0:
    op: x
    children: []
    eclass: "0"
    cost: 9223372036854776000
  y-0:
    op: y
    children: []
    eclass: "1"
    cost: 9223372036854776000
  i64-5871781006564002453:
    op: "1"
    children: []
    eclass: i64-5871781006564002453
    cost: 0
  distance-5871781006564002453:
    op: distance
    children:
      - Node-0
      - Node-5871781006564002453
    eclass: i64-5871781006564002453
    cost: 1
  a-0:
    op: a
    children: []
    eclass: "0"
    cost: 9223372036854776000
  b-0:
    op: b
    children: []
    eclass: "1"
    cost: 9223372036854776000
  i64-0:
    op: "0"
    children: []
    eclass: i64-0
    cost: 0
  Node-0:
    op: Node
    children:
      - i64-0
    eclass: "0"
    cost: 1
  Node-5871781006564002453:
    op: Node
    children:
      - i64-5871781006564002453
    eclass: "1"
    cost: 1
  i64-11743562013128004906:
    op: "2"
    children: []
    eclass: i64-11743562013128004906
    cost: 0
  Node-11743562013128004906:
    op: Node
    children:
      - i64-11743562013128004906
    eclass: "0"
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  Node-17615343019692007359:
    op: Node
    children:
      - i64-17615343019692007359
    eclass: "1"
    cost: 1
root_eclasses: []
class_data:
  "0":
    type: N
  "1":
    type: N
  i64-5871781006564002453:
    type: i64
  i64-0:
    type: i64
  i64-11743562013128004906:
    type: i64
  i64-17615343019692007359:
    type: i64

